File: Q29 print all duplicate character in a string .py
def printDuplicates(str):
    # Converting string to list of characters
    char_list = list(str)
    # Sorting the list of characters
    char_list.sort()
    
    # Loop through the sorted list to find duplicates
    i = 0
    while i < len(char_list):
        count = 1
        # Counting the occurrences of each character
        while i < len(char_list)-1 and char_list[i] == char_list[i+1]:
            count += 1
            i += 1
        
        # Printing the duplicate character and its count
        if count > 1:
            print(char_list[i], ", count = ", count)
        i += 1

File: Q 29 Print all duplicate character in a string.js
function printDuplicates(str) {
    let len = str.length;
    
    // Sorting the string
    str = str.split('').sort().join('');
    
    // Loop through the sorted string to find duplicates
    for(let i = 0; i < len; i++) {
        let count = 1;
        
        // Counting the occurrences of each character
        while(i < len-1 && str[i] == str[i+1]) {
            count++;
            i++;
        }
        
        // Printing the duplicate character and its count
        if(count > 1) {
            console.log(str[i] + ", count = " + count);
        }
    }
}
